capstone-api/models/User.js

```diff
@@ -308,6 +309,55 @@ class User {
     query.equalTo("username", username).descending("createdAt");
     return await query.find();
   }
+
+  static async getRecentSearches(username) {
+    const query = new Parse.Query("User");
+    query.equalTo("username", username);
+    return (await query.first({})).get("recentSearches");
+  }
+
+  static async addRecentSearch(username, searchValue) {
+    const query = new Parse.Query("User");
+    query.equalTo("username", username);
+    const user = await query.first({});
+
+    // logic to remove oldest search from queue - have to specifically remove the first element to get the first element to delete
+    let oldestSearch = "";
+    if (user.get("recentSearches").length >= 10) {
+      oldestSearch = user.get("recentSearches")[0];
+      user.remove("recentSearches", oldestSearch);
+    }
+
+    try {
+      await user.save();
+    } catch (error) {
+      return `Error with ${username} deleting oldest search value ${oldestSearch}`;
+    }
```

Comment:
I think you can put this in if (user.get("recentSearches").length >= 10)  block since it only should run if you need to delete an old search. Also you don't need to instantiate oldestSearch outside of the if section.